test/core/messages.integration.test.ts

```diff
@@ -352,6 +352,36 @@ describe('messages integration', () => {
       }),
     ]);
 
+    // Ensure that `current` pagination method works
+    const current = await history2?.current();
```

Comment:
For next, current and first - I think it would be a good idea to extend the test a bit to call a comparative method (e.g. before), to validate the changes.